### PR TITLE
Make sure basket is empty before coupon order

### PR DESCRIPTION
**Pull request by mjfrey**
Flush the basket before we create a coupon order.

### CODE REVIEW

**Comment by mjfrey** [comment]
@rlucioni

**Comment by rlucioni** [comment]
@mjfrey please add a regression test. Also, is there still a place for [`ecommerce.extensions.baskets.utils.prepare_basket()`](https://github.com/edx/ecommerce/blob/master/ecommerce/extensions/basket/utils.py#L11)? It performs similar operations, including flushing the basket.

**Comment by mjfrey** [comment]
@rlucioni Took your suggestion as this is better.

File: ecommerce/tests/mixins.py [hunk 1]
```diff
@@ -391,12 +392,11 @@ def create_coupon(
...
-            product=coupon,
-            client=factories.UserFactory(),
-            site=self.site,
-            partner=partner
-        )
+        request = RequestFactory()
+        request.site = self.site
+        request.user = factories.UserFactory()
+
+        self.basket = prepare_basket(request, coupon)
```
Comment (rlucioni) [inline ecommerce/tests/mixins.py:399]: @mjfrey this doesn't actually test the observed behavior, does it? We want to verify that the basket is flushed.

Comment (mjfrey) [inline ecommerce/tests/mixins.py:399]: There are already tests for prepare_basket that ensure a single item. They are in test_utils.py

Comment (rlucioni) [inline ecommerce/tests/mixins.py:399]: @mjfrey never mind, prepare_basket's tests cover this case: https://github.com/edx/ecommerce/blob/master/ecommerce/extensions/basket/tests/test_utils.py#L73

**Comment by rlucioni** [comment]
:+1:

**Comment by mjfrey** [comment]
@rlucioni not sure on how to merge since I am .01 coverage short.
Can you merge this?

**Comment by rlucioni** [comment]
@mjfrey I cannot, only @clintonb has that power :expressionless:. I'm looking around in our codecov settings to see if we can get the build to pass when coverage drops by small percentages like this. Coveralls allows it, I'm not sure about Codecov.

**Comment by rlucioni** [comment]
@mjfrey I changed a setting which should pass the build as long as coverage is > 95%. Try restarting your build.

**Comment by rlucioni** [comment]
Voila!

**Pull request merged**